### PR TITLE
all: upgrade Envoy source to v3alpha API.

### LINKED ISSUE #8082: v2 -> v3 API translation
- [x] Implement a v2 --> v3 translation code generator. This will take v2 inputs and re-express them as v3. As part of this, an internal only (not API visible) `deprecated_v2` field will be synthesized that can smuggle v2 fields into the mostly v3 Envoy internals.
- [ ] Update Envoy to accept v3 internally, using the above translator output for v2 inputs.

### PR DESCRIPTION
**Pull request by htuch**
This is the mega-upgrade patch taking Envoy from v2 to v3alpha internally. It consists of a mechanically generated upgrade patch ("Mechanical upgrade via api_boost.py.") and some post-upgrade manual fixups.

Risk level: Very high. It's possible that internal changes might manifest on the wire in unanticipated ways, e.g. via type URLs or config rejection.
Testing: All tests pass.

Fixes #8082

### CODE REVIEW

File: test/common/protobuf/utility_test.cc [hunk 1]
```diff
@@ -174,10 +174,10 @@ TEST_F(ProtobufUtilityTest, DowncastAndValidateUnknownFields) {
   envoy::config::bootstrap::v3alpha::Bootstrap bootstrap;
   bootstrap.GetReflection()->MutableUnknownFields(&bootstrap)->AddVarint(1, 0);
   EXPECT_THROW_WITH_MESSAGE(TestUtility::validate(bootstrap), EnvoyException,
-                            "Protobuf message (type envoy.config.bootstrap.v2.Bootstrap with "
+                            "Protobuf message (type envoy.config.bootstrap.v3alpha.Bootstrap with "
```
Comment (lizan) [inline test/common/protobuf/utility_test.cc]: Probably not actionable in this PR, but for future, do you think we should add a macro for these types in string, or do wild card matching like `envoy.*.Bootstrap`?

File: test/common/upstream/cluster_manager_impl_test.cc [hunk 2]
```diff
@@ -381,10 +381,10 @@ TEST_F(ClusterManagerImplTest, OriginalDstLbRestriction2) {
                   port_value: 11001
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(
-      create(parseBootstrapFromV2Yaml(yaml)), EnvoyException,
-      "cluster: LB policy ORIGINAL_DST_LB is not valid for Cluster type STATIC. "
-      "'ORIGINAL_DST_LB' is allowed only with cluster type 'ORIGINAL_DST'");
+  EXPECT_THROW_WITH_MESSAGE(create(parseBootstrapFromV2Yaml(yaml)), EnvoyException,
```
Comment (lizan) [inline test/common/upstream/cluster_manager_impl_test.cc]: shall we rename `parseBootstrapFromV2Yaml` to `parseBootstrapFromV3AlphaYaml`?

Comment (htuch) [reply test/common/upstream/cluster_manager_impl_test.cc]: I'll take an AI to do the cleanup in a follow PR, ideally we aim to keep this one as small as possible given the large mechanical base.

Comment (htuch) [reply test/common/upstream/cluster_manager_impl_test.cc]: (there are other methods to deal with here)

File: test/common/protobuf/utility_test.cc (hunk 1 above)
Comment (htuch) [reply test/common/protobuf/utility_test.cc]: I'll take this one as a followup, I agree it's a nice to have.

**Comment by htuch** [comment]
@envoyproxy/api-shepherds this should be ready-to-merge once CI passes and we have LGTMs.

**Comment by mattklein123** [comment]
(I didn't review this line by line but generally looks sane. Amazing work!. Let's merge and see what happens.)

**Pull request merged**